src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1567,25 +1576,27 @@ export default class Blockchain extends Emittery.Typed<
    * Gracefully shuts down the blockchain service and all of its dependencies.
    */
   public async stop() {
-    // If the blockchain is still initalizing we don't want to shut down
+    // If the blockchain is still initializing we don't want to shut down
     // yet because there may still be database calls in flight. Leveldb may
     // cause a segfault due to a race condition between a db write and the close
     // call.
     if (this.#state === Status.starting) {
       await this.once("ready");
     }
 
+    this.#state = Status.stopping;
+
     // stop the polling miner, if necessary
     clearTimeout(this.#timer);
 
     // clean up listeners
-    this.vm.removeAllListeners();
+    this.vm && this.vm.removeAllListeners();
```

Comment:
Since we now call `stop` internally on initialization exceptions we might not have everything ready, so we check for things' existence before trying to clean them up.